src/api_server/src/lib.rs

```diff
@@ -71,20 +79,73 @@ impl ApiServer {
     pub fn new(
         config: Config,
         sockets: Vec<SocketAddr>,
+        private_key: Option<PathBuf>,
+        certificate: Option<PathBuf>,
+        insecure: bool,
         attestation_service: Arc<AttestationService>,
         http_timeout: i64,
-    ) -> Self {
-        ApiServer {
+    ) -> Result<Self> {
+        if !insecure && (private_key.is_none() || certificate.is_none()) {
+            return Err(anyhow!("Missing HTTPS credentials"));
+        }
+
+        Ok(ApiServer {
             config,
             sockets,
+            private_key,
+            certificate,
+            insecure,
             attestation_service,
             http_timeout,
-        }
+        })
+    }
+
+    fn tls_config(&self) -> Result<ServerConfig> {
+        let cert_file = &mut BufReader::new(
+            File::open(
+                self.certificate
+                    .clone()
+                    .ok_or(anyhow!("Missing certificate"))?,
+            )
+            .map_err(anyhow::Error::from)?,
+        );
+
+        let key_file = &mut BufReader::new(
+            File::open(
+                self.private_key
+                    .clone()
+                    .ok_or(anyhow!("Missing private key"))?,
+            )
+            .map_err(anyhow::Error::from)?,
+        );
+
+        let cert_chain = certs(cert_file)
+            .map_err(anyhow::Error::from)?
```

Comment:
Same as mentioned. Could we directly use `?` to replace `.map_err(anyhow::Error::from)?` ?